Add questOwner to webhook.js

PitiTheGrey opened the pull request:
Adds questOwner:UUID to the json response of a webhook

[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)
This is the discussion:
https://github.com/HabitRPG/habitica/issues/12931
### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)



[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 9226cae6-d852-45da-a51b-ec2c9cb759e6

PitiTheGrey committed:
Add questOwner to webhook.js

Adds questOwner:UUID to the json response of a webhook

PitiTheGrey commented:
Please excuse me. I am just curious if there is any progress here.

SabreCat commented:
This should work fine, but could we add this data field to the expectations for webhook behavior in the relevant integration tests? See e.g. https://github.com/HabitRPG/habitica/blob/develop/test/api/v3/integration/quests/POST-groups_groupId_quests_invite.test.js#L238-L241

SabreCat reviewed (changes requested):
Needs some simple test expectation updates, otherwise good to go

PitiTheGrey committed:
Update POST-groups_groupId_quests_invite.test.js

Adds test for questOwner. Not sure, but I hope changes that have taken place in the meantime will not be undone by this. At least a recent commit about "send webhook to the quest inviter" is missing in my code.

PitiTheGrey commented:
Not sure if my commit conflicts with https://github.com/HabitRPG/habitica/pull/13111 
from 2 hours ago?

SabreCat commented:
> Not sure if my commit conflicts with #13111 from 2 hours ago?

Nope, that PR has to do with _when_ we send the webhook, while yours changes the _content_ of the webhook, so there's not a conflict!

That said, the new test appears to be failing. Does the `quest` object the webhook receives actually contain a value for `leader`?

PitiTheGrey committed:
Update webhook.js

PitiTheGrey commented:
Please delete in favour of #13277
